src/python/WMCore/WMSpec/StdSpecs/TaskChain.py

```diff
@@ -242,10 +242,18 @@ def __call__(self, workloadName, arguments):
             parentTask = None
             if parent in self.mergeMapping:
                 parentTask = self.mergeMapping[parent][parentTaskModule(taskConf)]
-
+                
             task = self.makeTask(taskConf, parentTask)
-            #sets the prepID for the task (overwrite initial PrepID if it is set by workflow
-            task.setPrepID(taskConf["PrepID"])
+            
+            
+            # sets the prepID  all the properties need to be set by
+            # self.workload.setTaskPropertiesFromWorkload manually for the task
+            task.setPrepID(taskConf.get("PrepID", self.workload.getPrepID()))
+            task.setAcquisitionEra(taskConf.get("AcquisitionEra", self.workload.acquisitionEra))
+            task.setProcessingString(taskConf.get("ProcessingString", self.workload.processingString))
+            task.setProcessingVersion(taskConf.get("ProcessingString", self.workload.processingVersion))
+            task.setLumiMask(taskConf.get("LumiList", self.workload.lumiList))
```

Comment:
setupLumiMask do nothing if mask is {} but I guess we can check before that to save the unnecessary function call.
 
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMSpec/WMTask.py#L1246

But I was wondering whether this need to set for both setupGeneratorTask and setupTask? (comment below)